### PR TITLE
Fix decimal expression can't reuse

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/CastOperator.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorsReuseRule.java|1|1|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/ExpressionTest.java [hunk 1]
```diff
@@ -963,4 +964,12 @@ public void testReduceNumberCast() throws Exception {
         plan = getFragmentPlan(sql);
         assertContains(plan, "CAST(11: id_bool AS VARCHAR(65533))");
     }
+
+    @Test
+    public void testDecimalReuse() throws Exception {
+        String sql = "select id_decimal + 1, id_decimal + 2 from test_all_type";
+        String plan = getFragmentPlan(sql);
+        assertContains(plan, "common expressions:\n" +
+                "  |  <slot 13> : CAST(10: id_decimal AS DECIMAL64(18,2))");
```
Comment (HangyuanLiu) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/ExpressionTest.java:973]: varchar equal is right?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/CastOperator.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorsReuseRule.java|1|1|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**